Add site risk score column to group overview

lauramaxwell opened the pull request.

samussiah commented on examples/data/results.csv:
```diff
@@ -113570,3 +113570,190 @@ AA-AA-000-0000,Site,kri0012,48,0,1,0,,,01.12.2019
 AA-AA-000-0000,Site,kri0012,55,0,1,0,,,01.12.2019
 AA-AA-000-0000,Study,qtl0004,AA-AA-000-0000,478,984366,0.000485592,0.00044303,0,01.12.2019
 AA-AA-000-0000,Study,qtl0006,AA-AA-000-0000,89,1301,0.068408916,0.055294789,0,01.12.2019
+AA-AA-000-0000,Site,srs0001,99,0,50,0,0,0,01.12.2019
```
huh, snapshot dates got re-formatted at some point

samussiah commented:
The site example isn't currently working.  Pretty it's only because the metric is missing in the data.  In that scenario let's just disable the column entirely.

<img width="1023" height="477" alt="image" src="https://github.com/user-attachments/assets/828bfb53-3427-484f-b5c2-7308f0aecf9e" />

samussiah commented on src/groupOverview/makeTable/addCustomTooltip.js:
```diff
@@ -0,0 +1,114 @@
+    const riskScoreCells = cells.filter((d) => d.column.valueKey === 'siteRiskScore' && d.tooltip);
+
+    riskScoreCells
+        .on('mouseenter', function(event, d) {
```
There's some conflict between this and some other event listener.  Not sure if it's the row or the cell itself, but the tooltip disappears moving within the cell.

lauramaxwell replied:
Looks like it's the row highlight mouse-over listener that is causing the issue. I don't want to get rid of the row highlight feature in the Risk Score column, so I'm thinking of making it a click event to see the details of the risk score. thoughts, @samussiah @jwildfire?

samussiah commented on src/groupOverview/makeTable/addCustomTooltip.js:
Love the new tooltip!

samussiah commented on src/groupOverview/defineColumns/defineGroupColumns.js:
```diff
@@ -73,5 +76,37 @@ export default function defineGroupColumns(groupMetadata, config) {
+        metricMetadatum.GroupLevel === 'Site'
+    );
+    
+    if (config.GroupLevel === 'Site' && hasSiteLevelData &&
```
Let's attach this check to the config.  The check should verify that the results contain site risk score rows, and if so, add the column.

samussiah commented on src/groupOverview/defineColumns/defineRiskScoreTooltip.js:
```diff
@@ -0,0 +1,84 @@
+            const metricName = metricInfo ? 
+                `${metricInfo.abbreviation} - ${metricInfo.name}` : 
+                result.MetricID;
+            tooltipLines.push(`• ${metricName}: ${result.Weight}`);
```
need to add data with new columns

samussiah replied:
actually, this should just be `Numerator`, right?

samussiah reviewed (changes requested):
Love the new tooltip, and :

- add a data-driven configuration setting that verifies we're working with site-level data and that site risk score exists, and use this to enable the SRS column
- figure out the hover event conflict that makes the new tooltip disappear hovering within the cell
- pull the link to more information on SRS out into a configuration setting
- did the team decide not to add a color scale on SRS?

lauramaxwell commented:
> Love the new tooltip, really nice work @lauramaxwell! A few updates:
> 
> * add a data-driven configuration setting that verifies we're working with site-level data and that site risk score exists, and use this to enable the SRS column
> * figure out the hover event conflict that makes the new tooltip disappear hovering within the cell
> * pull the link to more information on SRS out into a configuration setting
> * did the team decide not to add a color scale on SRS?

Ok, I'll get to work on these. Just to clarify on the third point- do you want the link to be able to be set from the Widget call in R? or just moved into the config file?

And, re: color scale- yes, the team decided not to have a color scale- at least until it is better understood what is a "low" or "high" risk score.

samussiah commented:
> Ok, I'll get to work on these. Just to clarify on the third point- do you want the link to be able to be set from the Widget call in R? or just moved into the config file?

No explicit R integration, just a new setting named `SiteRiskDetails` or something.  It could be an array of text that appears in lines in the tooltip.  Worried if that link ever changes.

lauramaxwell commented:
@samussiah ready for re-review. I think I addressed all of the issues you brought up, assuming a click event for the tooltip is acceptable. You'll have to take a look at gsm.kri to actually see examples of the tooltip at work, or see the attached file.
[kri_report_AAAA0000000_Site_20250905.html](https://github.com/user-attachments/files/22175592/kri_report_AAAA0000000_Site_20250905.html)

lauramaxwell commented:
> @samussiah ready for re-review. I think I addressed all of the issues you brought up, assuming a click event for the tooltip is acceptable. You'll have to take a look at gsm.kri to actually see examples of the tooltip at work, or see the attached file. [kri_report_AAAA0000000_Site_20250905.html](https://github.com/user-attachments/files/22175592/kri_report_AAAA0000000_Site_20250905.html)

@jwildfire

jwildfire commented:
@lauramaxwell Click event doesn't seem to trigger on rows that are hidden on initial load. Try changing the select from "1+ Red Flag" to "All" and clicking a cell ...

lauramaxwell commented:
> @lauramaxwell Click event doesn't seem to trigger on rows that are hidden on initial load. Try changing the select from "1+ Red Flag" to "All" and clicking a cell ...

Oh good catch- i'll see what i can do to fix that so the tooltip is not dependent on visibility on initial load

lauramaxwell commented:
@jwildfire i made a fix and pushed it [here](https://github.com/Gilead-BioStats/gsm.viz/pull/454/commits/4f4ab44525ad54170e1cb5fa5e0de76f7def3786). Looks like i had forgotten to add the custom tooltip object to the `updateTable` script that updates the table when the view changes. looks better to me now. here's the gsm.kri sample output with these edits: [kri_report_AAAA0000000_Site_20250911.html](https://github.com/user-attachments/files/22281260/kri_report_AAAA0000000_Site_20250911.html)

jwildfire commented:
Working for me. I'm going to go ahead and approve.

lauramaxwell commented:
@samussiah - can you mark requested changes as approved if you have a moment- Jeremy was able to take a look and provide some final feedback
